### DISCUSSION TITLE
Hard Limit Problem with MKS Gen V1.4 + GRBL 5X

### DISCUSSION THREAD

**Discussion by jafarniknam** [fra589/grbl-Mega-5X#247] [Q&A] [answered]
Hello .
I am using MKS Gen V1.4 and GRBL mega 5x board for CNC machine and I used openBuild Control software to set it up.
For each axis, I installed two microswitches at the beginning and end and activated the Hard limits. The Homing cycle works well and becomes a successful Home. In the Troubleshooting section, changes of microswitches can also be seen.
The problem is that in the Jog section, the axes do not show any specific error when they hit the microswitch and continue to move. While in the original version, GRBL with CNC Shield + Uno board stops when it hits the microswitches and an error appears on the screen. Thanks if anyone can help

**Comment by fra589** [comment, marked as answer]
Hi @jafarniknam,  
  
Have you made just a little search on the web?  
https://www.google.com/search?q=grbl-Mega-5X+hardware+limits  
  
Hardware limits are disables by default in config.h for performance optimization.  
Did you enabled it by uncommenting the line `#define ENABLE_RAMPS_HW_LIMITS` in cpu_map.h?  
  
@++;  
Gauthier.

**Reply by jafarniknam** [reply]
Hello . Thank you very much for your prompt reply. There was a problem disabling ENABLE_RAMPS_HW_LIMITS that I did not know existed. The problem was solved

**Comment by jafarniknam** [comment]
@fra589 
I also do not know which pin to connect for the emergency stop button. Can you help in this case too? There are pins for Abort, Hold, Resume, E-Stop on the CNC shield. In the cpu_map.h file all these pins are available except E_Stop . 


  // Define user-control CONTROLs (cycle start, reset, feed hold) input pins.
  // NOTE: All CONTROLs pins must be on the same port and not on a port with other input pins (limits).
  #define CONTROL_DDR       DDRK
  #define CONTROL_PIN       PINK
  #define CONTROL_PORT      PORTK
  #define CONTROL_RESET_BIT         1  // Pin A9 - RAMPS Aux 2 Port
  #define CONTROL_FEED_HOLD_BIT     2  // Pin A10 - RAMPS Aux 2 Port
  #define CONTROL_CYCLE_START_BIT   3  // Pin A11 - RAMPS Aux 2 Port
  #define CONTROL_SAFETY_DOOR_BIT   4  // Pin A12 - RAMPS Aux 2 Port
  #define CONTROL_INT       PCIE2  // Pin change interrupt enable pin
  #define CONTROL_INT_vect  PCINT2_vect
  #define CONTROL_PCMSK     PCMSK2 // Pin change interrupt register
  #define CONTROL_MASK      ((1<<CONTROL_RESET_BIT)|(1<<CONTROL_FEED_HOLD_BIT)|(1<<CONTROL_CYCLE_START_BIT)|(1<<CONTROL_SAFETY_DOOR_BIT))
 

Thanks .

**Reply by fra589** [reply]
Hi @jafarniknam,  
  
You can connect your hardware emergency button to the reset pin.  
On my personal machine, I use a button with 2 circuit (1NO + 1NC): One normally open connected to the reset pin and the second normally closed which cut the power of motors.
https://fr.aliexpress.com/item/32672562816.html?gatewayAdapt=glo2fra&spm=a2g0o.9042311.0.0.1b636c37HHb5GT
  
@++;  
Gauthier.
